thingif/src/main/java/com/kii/thingif/query/Aggregation.java

```diff
@@ -0,0 +1,128 @@
+package com.kii.thingif.query;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Represent aggregation
+ */
+public class Aggregation implements Parcelable{
+    private @NonNull final FunctionType function;
+    private @NonNull final String field;
+    private @NonNull final FieldType fieldType;
+
+    public static enum FunctionType{
+        MAX,
+        MIN,
+        MEAN,
+        SUM
+    }
+
+    public static enum FieldType{
+        INTEGER,
+        DECIMAL
+    }
+
+    /**
+     * Initialize Aggregation
+     * @param function Agrregation function.
+     * @param field
+     */
+    private Aggregation(
+            @NonNull FunctionType function,
+            @NonNull String field,
+            @NonNull FieldType fieldType
+    ) {
+        this.function = function;
+        this.field = field;
+        this.fieldType = fieldType;
+    }
+
+
+    public  static Aggregation newMaxAggregation(
```

Comment:
We separates `count` and other functions. As a result, We do not need factory methods. Because we can use  `FieldType.INTEGER` and `FieldType.DECIMAL` for all of `Aggregate.FunctionType`.

What do you think?